pkg/kubectl/cmd/util/factory.go

```diff
@@ -157,6 +158,7 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 
 	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
 	flags.SetNormalizeFunc(util.WarnWordSepNormalizeFunc) // Warn for "_" flags
+	flags.Bool("disable-dynamic-discovery", false, "If true, disable calls to dynamically discover dynamic APIs")
```

Comment:
I'm still concerned about the behavior with regard to the number of requests for each command.  Also, if we're going to gate this code instead of having a separate method, passing a `bool` through the method arguments is much easier to compose.

I think we should try to avoid negative logic in flag names.

Adding @kubernetes/kubectl for additional comments on the behavior/name and @smarterclayton since he's concerned about the number of requests on client commands.